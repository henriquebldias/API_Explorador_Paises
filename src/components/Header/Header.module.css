.header {
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.headerDark {
    background-color: #1F2937;
    color: #fff;
}

.headerLight {
    background-color: #fff;
    color: #1F2937;
}

.container {
    max-width: 1280px;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 24px 16px;
}

.title {
    font-size: 20px;
    font-weight: 700;
    text-decoration: none;
    color: inherit;
    cursor: pointer;
}

.themeButton {
    background: none;
    border: none;
    display: flex;
    align-items: center;
    gap: 8px;
    color: inherit;
    cursor: pointer;
}

