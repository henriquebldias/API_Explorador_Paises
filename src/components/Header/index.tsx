import React from "react";
import { Link } from "react-router-dom";
import styles from "./Header.module.css";
import {Moon, Sun} from "lucide-react";

interface HeaderProps {
    isDarkMode: boolean
    toogleDarkMode: () => void
}

const Header: React.FC<HeaderProps> = ({isDarkMode, toogleDarkMode}) => {
    return (
        <header className={`${styles.header} ${isDarkMode ? styles.headerDark: styles.headerLight}`}>
            <div className={styles.container}>
                <Link className={styles.title} to="/">Onde no mundo?</Link>
                <button className={styles.themeButton} onClick={toogleDarkMode}>
                    {isDarkMode ? <Sun size={20}/>:<Moon size={20}/>}
                    <span>{isDarkMode ? "Modo Claro" : "Modo Escuro"}</span>
                </button>
            </div>
        </header>
    )
}

export default Header