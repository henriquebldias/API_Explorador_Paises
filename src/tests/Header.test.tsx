import {describe, it, expect, vi} from "vitest"
import {fireEvent, render, screen} from "@testing-library/react"
import {MemoryRouter} from "react-router-dom"
import Header from "../components/Header"

describe('Header tests', () => {
    it('Deve renderizar texto do botão de alterar cor do tema no modo claro', () => {
        const mockToogle = vi.fn()
        
        const testData = render(
            <MemoryRouter>
                <Header 
                    isDarkMode={false}
                    toogleDarkMode={mockToogle}
                />
            </MemoryRouter> 
        )

        expect(screen.getByText('Onde no mundo?')).toBeDefined()
        expect(screen.getByText('Modo Escuro')).toBeDefined()
        expect(testData).toMatchSnapshot()
    })

    it('Deve renderizar texto do botão de alterar cor do tema no modo escuro', () => {
        const mockToogle = vi.fn()
        
        const testData = render(
            <MemoryRouter>
                <Header 
                    isDarkMode={true}
                    toogleDarkMode={mockToogle}
                />
            </MemoryRouter> 
        )

        expect(screen.getByText('Onde no mundo?')).toBeDefined()
        expect(screen.getByText('Modo Claro')).toBeDefined()
        expect(testData).toMatchSnapshot()
    })

    it('Deve aplicar headerLight quando isDarkMode === false (modo claro)', () => {
        const mockToogle = vi.fn()

        const testData = render(
            <MemoryRouter>
                <Header 
                    isDarkMode={false}
                    toogleDarkMode={mockToogle}
                />
            </MemoryRouter> 
        )

        const header = screen.getByText('Onde no mundo?').closest('header')

        expect(header?.className).toContain('_headerLight_63c2ea')
        expect(testData).toMatchSnapshot()
    })

    it('Deve aplicar headerDark quando isDarkMode === true (modo escuro)', () => {
        const mockToogle = vi.fn()

        const testData = render(
            <MemoryRouter>
                <Header 
                    isDarkMode={true}
                    toogleDarkMode={mockToogle}
                />
            </MemoryRouter> 
        )

        const header = screen.getByText('Onde no mundo?').closest('header')

        expect(header?.className).toContain('_headerDark_63c2ea')
        expect(testData).toMatchSnapshot()
    })

    it('Chama função toggleDarkMode quando botão de tema é clicado no modo claro', () => {
        const mockToogle = vi.fn()
        
        const testData = render(
            <MemoryRouter>
                <Header 
                    isDarkMode={false}
                    toogleDarkMode={mockToogle}
                />
            </MemoryRouter> 
        )

        fireEvent.click(screen.getByText('Modo Escuro'))

        expect(mockToogle).toHaveBeenCalled()
        expect(testData).toMatchSnapshot()
    })

    it('Chama função toggleDarkMode quando botão de tema é clicado no modo escuro', () => {
        const mockToogle = vi.fn()
        
        const testData = render(
            <MemoryRouter>
                <Header 
                    isDarkMode={true}
                    toogleDarkMode={mockToogle}
                />
            </MemoryRouter> 
        )

        fireEvent.click(screen.getByText('Modo Claro'))

        expect(mockToogle).toHaveBeenCalled()
        expect(testData).toMatchSnapshot()
    })
})