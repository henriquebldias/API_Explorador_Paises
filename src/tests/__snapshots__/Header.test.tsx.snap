// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header tests > Chama função toggleDarkMode quando botão de tema é clicado no modo claro 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_63c2ea _headerLight_63c2ea"
      >
        <div
          class="_container_63c2ea"
        >
          <a
            class="_title_63c2ea"
            href="/"
          >
            Onde no mundo?
          </a>
          <button
            class="_themeButton_63c2ea"
          >
            <svg
              class="lucide lucide-moon "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
            <span>
              Modo Escuro
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_63c2ea _headerLight_63c2ea"
    >
      <div
        class="_container_63c2ea"
      >
        <a
          class="_title_63c2ea"
          href="/"
        >
          Onde no mundo?
        </a>
        <button
          class="_themeButton_63c2ea"
        >
          <svg
            class="lucide lucide-moon "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span>
            Modo Escuro
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header tests > Chama função toggleDarkMode quando botão de tema é clicado no modo escuro 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_63c2ea _headerDark_63c2ea"
      >
        <div
          class="_container_63c2ea"
        >
          <a
            class="_title_63c2ea"
            href="/"
          >
            Onde no mundo?
          </a>
          <button
            class="_themeButton_63c2ea"
          >
            <svg
              class="lucide lucide-sun "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
            <span>
              Modo Claro
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_63c2ea _headerDark_63c2ea"
    >
      <div
        class="_container_63c2ea"
      >
        <a
          class="_title_63c2ea"
          href="/"
        >
          Onde no mundo?
        </a>
        <button
          class="_themeButton_63c2ea"
        >
          <svg
            class="lucide lucide-sun "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <span>
            Modo Claro
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header tests > Deve aplicar headerDark quando isDarkMode === true (modo escuro) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_63c2ea _headerDark_63c2ea"
      >
        <div
          class="_container_63c2ea"
        >
          <a
            class="_title_63c2ea"
            href="/"
          >
            Onde no mundo?
          </a>
          <button
            class="_themeButton_63c2ea"
          >
            <svg
              class="lucide lucide-sun "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
            <span>
              Modo Claro
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_63c2ea _headerDark_63c2ea"
    >
      <div
        class="_container_63c2ea"
      >
        <a
          class="_title_63c2ea"
          href="/"
        >
          Onde no mundo?
        </a>
        <button
          class="_themeButton_63c2ea"
        >
          <svg
            class="lucide lucide-sun "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <span>
            Modo Claro
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header tests > Deve aplicar headerLight quando isDarkMode === false (modo claro) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_63c2ea _headerLight_63c2ea"
      >
        <div
          class="_container_63c2ea"
        >
          <a
            class="_title_63c2ea"
            href="/"
          >
            Onde no mundo?
          </a>
          <button
            class="_themeButton_63c2ea"
          >
            <svg
              class="lucide lucide-moon "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
            <span>
              Modo Escuro
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_63c2ea _headerLight_63c2ea"
    >
      <div
        class="_container_63c2ea"
      >
        <a
          class="_title_63c2ea"
          href="/"
        >
          Onde no mundo?
        </a>
        <button
          class="_themeButton_63c2ea"
        >
          <svg
            class="lucide lucide-moon "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span>
            Modo Escuro
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header tests > Deve renderizar texto do botão de alterar cor do tema no modo claro 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_63c2ea _headerLight_63c2ea"
      >
        <div
          class="_container_63c2ea"
        >
          <a
            class="_title_63c2ea"
            href="/"
          >
            Onde no mundo?
          </a>
          <button
            class="_themeButton_63c2ea"
          >
            <svg
              class="lucide lucide-moon "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
            <span>
              Modo Escuro
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_63c2ea _headerLight_63c2ea"
    >
      <div
        class="_container_63c2ea"
      >
        <a
          class="_title_63c2ea"
          href="/"
        >
          Onde no mundo?
        </a>
        <button
          class="_themeButton_63c2ea"
        >
          <svg
            class="lucide lucide-moon "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span>
            Modo Escuro
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header tests > Deve renderizar texto do botão de alterar cor do tema no modo escuro 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_63c2ea _headerDark_63c2ea"
      >
        <div
          class="_container_63c2ea"
        >
          <a
            class="_title_63c2ea"
            href="/"
          >
            Onde no mundo?
          </a>
          <button
            class="_themeButton_63c2ea"
          >
            <svg
              class="lucide lucide-sun "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
            <span>
              Modo Claro
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_63c2ea _headerDark_63c2ea"
    >
      <div
        class="_container_63c2ea"
      >
        <a
          class="_title_63c2ea"
          href="/"
        >
          Onde no mundo?
        </a>
        <button
          class="_themeButton_63c2ea"
        >
          <svg
            class="lucide lucide-sun "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <span>
            Modo Claro
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
